### ISSUE TITLE
Config file created incorrectly when using multiple values in rke2_kubelet_args

### ISSUE THREAD

**Issue by Yantrio** [rancherfederal/rke2-ansible#60]
If I consume the playbook and set the following vars:
```yaml
vars:
    rke2_kubelet_args:
      - "feature-gates=DynamicKubeletConfig=false"
      - "image-gc-high-threshold=100"
      - "image-gc-low-threshold=99"
 ```
 
 Then the config file that is generated has multiple instances of the `kubelet-arg: ` variable set, instead of 1 single instance set with an array of values.
 
 Expected output:
 ```yaml
kubelet-arg: 
  - feature-gates=DynamicKubeletConfig=false
  - image-gc-high-threshold=100
  - image-gc-low-threshold=99
  ```
  
  Actual:
 ```yaml
 kubelet-arg: feature-gates=DynamicKubeletConfig=false
kubelet-arg: image-gc-high-threshold=100
kubelet-arg: image-gc-low-threshold=99
 ```
 
 In my specific case, kubelet then fails to load as it only knows about `image-gc-low-threshold` and not `image-gc-high-threshold` so I think this could've easily been missed in other clusters.
 
 
 Offending task:
 https://github.com/rancherfederal/rke2-ansible/blob/bb0e7a1f7016ba73fde7a457e7d4246e131a06d7/roles/rke2_common/tasks/config.yml#L46-L51
 
 
 EDIT:
 
 I also assume this is the case for the other array based vars:
 - kube-apiserver-arg
 - kube-scheduler-arg
 - kube-controller-manager-arg
 - kubelet-arg
 - node-label

**Comment by Yantrio** [comment]
I can put in a PR to fix this, but I'm not sure if there's a specific reason it was implemented the way that it was.

At the moment I'm working around this by using 
```yaml
- name: Add rke2_kubelet_args
  lineinfile:
    path: /etc/rancher/rke2/config.yaml
    line: "kubelet-arg: {{ rke2_kubelet_args | default([]) | to_yaml(width=9999999) | trim }}"
```

This print the array as a single line like so:
```yaml
kubelet-arg: [feature-gates=DynamicKubeletConfig=false, image-gc-high-threshold=100, image-gc-low-threshold=99]
```

But I feel like this is a hacky workaround, I would prefer for it to be the same as the expected output in my original comment.
    
I assume its best to wait for the refactoring PR to go in because I also noticed that these tasks are removed: https://github.com/rancherfederal/rke2-ansible/pull/54/files

@mddamato was there a reason they were removed in there?

**Comment by mddamato** [comment]
yes @Yantrio there was a bug for some time in rke2 where the only way to add multiple args for things like the kubelet was to have multiple lines in the config file as shown in your "Actual". This was recently fixed and this code has been replaced already we're just testing to make sure it works ok. We should be merging in #54 which should fix this problem today.

**Comment by Yantrio** [comment]
That sounds great. Thanks for the quick reply and I'll switch over to that branch for now until the PR is merged. :+1:

**Comment by mddamato** [comment]
fixed in #54
